app/src/main/java/com/smartpack/kernelmanager/utils/kernel/misc/qtihaptics.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2015-2016 Willi Ye <williye97@gmail.com>
+ *
+ * This file is part of Kernel Adiutor.
+ *
+ * Kernel Adiutor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Kernel Adiutor is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Kernel Adiutor.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.smartpack.kernelmanager.utils.kernel.misc;
+
+import android.content.Context;
+
+import com.smartpack.kernelmanager.fragments.ApplyOnBootFragment;
+import com.smartpack.kernelmanager.utils.Utils;
+import com.smartpack.kernelmanager.utils.root.Control;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Created by willi on 29.06.16.
+ */
+public class Vibration {
+
+    private static Vibration sInstance;
+
+    public static Vibration getInstance() {
+        if (sInstance == null) {
+            sInstance = new Vibration();
+        }
+        return sInstance;
+    }
+
+    private static final String VIB_LIGHT = "/sys/module/qti_haptics/parameters/vmax_mv_override";
+    private static final String VIB_ENABLE = "/sys/module/qti_haptics/parameters/vmax_mv_enable";
+
+    private final HashMap<String, MinMax> mVibrations = new HashMap<>();
```

Comment:
For the implementation, I believe the other entries must be preserved on the list.